lib/src/vector_math/matrix4.dart

```diff
@@ -864,27 +982,65 @@ class Matrix4 {
     _m4storage[7] = t8;
   }
 
-  /// Scale this matrix by a [Vector3], [Vector4], or x,y,z
+  /// Scale this matrix by a [Vector3], [Vector4], or x,y,z as [double]s.
+  ///
+  /// If you know the argument types in a call site, prefer [scaleByDouble],
+  /// [scaleByVector3], or [scaleByVector4] for performance.
+  @pragma('wasm:prefer-inline')
+  @pragma('vm:prefer-inline')
```

Comment:
This function doesn't have multiple returns in it, so dart2js would inline it. Why not add `@pragma('dart2js:prefer-inline')` here as well?

(As mentioned in other comment, I think we should convert all functions to have only a single return, allowing dart2js optimizing all of them)